examples/basic/ipc_shm_ipcapi_consumer.c

```diff
@@ -0,0 +1,229 @@
+
+    // listen for the producer
+    if (listen(consumer_socket, 1) < 0) {
+        fprintf(stderr, "[consumer] ERROR: listen() failed\n");
+        goto err_close_consumer_socket;
+    }
+
+    fprintf(stderr, "[consumer] Listening for incoming connections ...\n");
+
+    // accept an incoming connection
+    producer_addr_len = sizeof(producer_addr);
+    producer_socket = accept(consumer_socket, (struct sockaddr *)&producer_addr,
+                             (socklen_t *)&producer_addr_len);
+    if (producer_socket < 0) {
+        fprintf(stderr, "[consumer] ERROR: accept() failed\n");
+        goto err_close_consumer_socket;
+    }
+
+    fprintf(stderr, "[consumer] Producer connected at IP %s and port %i\n",
+            inet_ntoa(producer_addr.sin_addr), ntohs(producer_addr.sin_port));
+
+    ret = producer_socket; // success
+
+err_close_consumer_socket:
+    close(consumer_socket);
+
+    return ret;
+}
+
+int main(int argc, char *argv[]) {
+    char consumer_message[CONSUMER_MSG_SIZE];
+    char recv_buffer[RECV_BUFF_SIZE];
+    int producer_socket;
+    int ret = -1;
+
+    if (argc < 2) {
+        fprintf(stderr, "usage: %s port\n", argv[0]);
+        return -1;
+    }
+
+    int port = atoi(argv[1]);
+
+    umf_memory_provider_handle_t OS_memory_provider = NULL;
+    umf_os_memory_provider_params_t os_params;
+    enum umf_result_t umf_result;
+
+    os_params = umfOsMemoryProviderParamsDefault();
+    os_params.visibility = UMF_MEM_MAP_SHARED;
+
+    // create OS memory provider
+    umf_result = umfMemoryProviderCreate(umfOsMemoryProviderOps(), &os_params,
+                                         &OS_memory_provider);
+    if (umf_result != UMF_RESULT_SUCCESS) {
+        fprintf(stderr,
+                "[consumer] ERROR: creating OS memory provider failed\n");
+        return -1;
+    }
+
+    umf_memory_pool_handle_t scalable_pool;
+    umf_result = umfPoolCreate(umfScalablePoolOps(), OS_memory_provider, NULL,
+                               0, &scalable_pool);
+    if (umf_result != UMF_RESULT_SUCCESS) {
+        fprintf(stderr,
+                "[producer] ERROR: creating jemalloc UMF pool failed\n");
+        goto err_destroy_OS_memory_provider;
+    }
+
+    producer_socket = consumer_connect(port);
+    if (producer_socket < 0) {
+        goto err_destroy_scalable_pool;
+    }
+
+    // receive the IPC handle from the producer's
+    ssize_t len = recv(producer_socket, recv_buffer, RECV_BUFF_SIZE, 0);
+    if (len < 0) {
+        fprintf(stderr, "[consumer] ERROR: recv() failed\n");
+        goto err_close_producer_socket;
+    }
+
+    umf_ipc_handle_t IPC_handle = (umf_ipc_handle_t)recv_buffer;
+
+    fprintf(
+        stderr,
+        "[consumer] Received the IPC handle from the producer (%zi bytes)\n",
+        len);
+
+    void *SHM_ptr;
+    umf_result = umfOpenIPCHandle(scalable_pool, IPC_handle, &SHM_ptr);
+    if (umf_result == UMF_RESULT_ERROR_NOT_SUPPORTED) {
+        fprintf(stderr,
```

Comment:
why do we need special processing for the `UMF_RESULT_ERROR_NOT_SUPPORTED` ? 
Furthermore, this block of code is not reachable, because if `umfOpenIPCHandle` is not supported then `umfGetIPCHandle` is not supported as well by the same memory provider on the producer side (the producer is not able to send the IPC handle in that case).

Reply:
We need this to correctly handle the `UMF_RESULT_ERROR_NOT_SUPPORTED` error, because `umfOpenIPCHandle()` can return `UMF_RESULT_ERROR_NOT_SUPPORTED` when `! defined(__NR_pidfd_open) || ! defined(__NR_pidfd_getfd)`.
It is reachable on Ubuntu 20.04 where the `! defined(__NR_pidfd_open) || ! defined(__NR_pidfd_getfd)` condition is true:
https://github.com/ldorau/unified-memory-framework/actions/runs/9002739451/job/24731880245

Reply:
Ah, I see.